index.js

```diff
@@ -82,87 +74,51 @@ function createDecoder(decoderLayers) {
 }
 
 function buildModel() {
-  const autoencoder = tf.sequential();
-  // Build the model
 
-  // Encoder
-  autoencoder.add(
-    tf.layers.dense({
-      units: 256,
-      inputShape: [W * W],
-      activation: "relu",
-    })
-  );
-  autoencoder.add(
-    tf.layers.dense({
-      units: 128,
-      activation: "relu",
-    })
-  );
-  autoencoder.add(
-    tf.layers.dense({
-      units: 64,
-      activation: "relu",
-    })
-  );
-  autoencoder.add(
-    tf.layers.dense({
-      units: 16,
-      activation: "relu",
-    })
-  );
-  autoencoder.add(
-    tf.layers.dense({
-      units: 4,
-      activation: "sigmoid",
-    })
-  );
-  // How do I start from here?
-  // Decoder
-
-  let decoderLayers = [];
-  decoderLayers.push(
-    tf.layers.dense({
-      units: 16,
-      activation: "relu",
-    })
-  );
-  decoderLayers.push(
-    tf.layers.dense({
-      units: 64,
-      activation: "relu",
-    })
-  );
-  decoderLayers.push(
-    tf.layers.dense({
-      units: 128,
-      activation: "relu",
-    })
-  );
-  decoderLayers.push(
-    tf.layers.dense({
-      units: 256,
-      activation: "relu",
-    })
-  );
-  decoderLayers.push(
-    tf.layers.dense({
-      units: W * W,
-      activation: "sigmoid",
-    })
-  );
-
-  for (let layer of decoderLayers) {
-    autoencoder.add(layer);
-  }
 
-  // const learningRate = 0.001;
-  // const optimizer = tf.train.adam(learningRate);
+// encoder
+  const encoder_input = tf.input({shape: [W*W], name: "encoder_input"});
+ // const l0 = tf.layers.flatten().apply(encoder_input);
+  const l1 = tf.layers.dense({units: 128, activation: "relu"}).apply(encoder_input);
+  const l2 = tf.layers.dense({units: 64, activation: "relu"}).apply(l1);
+  const l3 = tf.layers.dense({units: 16, activation: "relu"}).apply(l2);
+  const l4 = tf.layers.dense({units: 4, activation: "relu"}).apply(l3);
+  let encoded = tf.layers.dense({units: 2, activation: "relu", name: "encoder_output"}).apply(l4);
+  
+  let encoder = tf.model({inputs: encoder_input, outputs: encoded, name: "encoder"});
```

Comment:
I haven't used `tf.model` before, is it required for this methodology, or can I keep `tf.sequential`? Is `tf.model` preferred?

Reply:
from [https://js.tensorflow.org/api/latest/#sequential](https://js.tensorflow.org/api/latest/#sequential)

> The key difference between tf.model() and tf.sequential() is that tf.model() is more generic, supporting an arbitrary graph (without cycles) of layers. tf.sequential() is less generic and supports only a linear stack of layers.